pandas/core/dtypes/concat.py

```diff
@@ -132,6 +183,73 @@ def is_nonempty(x) -> bool:
     return np.concatenate(to_concat, axis=axis)
 
 
+def concat_arrays(to_concat) -> ArrayLike:
+    """
+    Alternative for concat_compat but specialized for use in the ArrayManager.
```

Comment:
if we have different impl for array vs block manager then let's push this code down into those areas (so its localized). ok for a followup.

Reply:
the problem is the logic is too scattered and very hard to grok where things are happening.